### ISSUE TITLE
Rework

### ISSUE THREAD

**Issue by sheridanchris** [fsharpforyou/fsharpforyou.github.io#30]
This needs to be reworked. I want to create a resource that teaches you F# while working on projects with a sensible iteration in language features/concepts and complexity. Right now, all this resource does is iterate through language F# concepts and explain them one by one... it's not very practical and you don't actually build _anything_. We should start slow with console applications and eventually progress to web applications, data science, and other concepts. 1 chapter per project and each of them should introduce new concepts / language features that weren't used in the last.

**Comment by TheAngryByrd** [comment]
>  We should start slow with console applications and eventually progress to web applications, data science, and other concepts. 1 chapter per project and each of them should introduce new concepts / language features that weren't used in the last.

I think here you would want to stick with one project type, such as building up a web app. Reason being, if you've discussed a few concepts that are a bit dense without explanation, jumping to the data science section might bounce off it.

**Comment by sheridanchris** [comment]
> > We should start slow with console applications and eventually progress to web applications, data science, and other concepts. 1 chapter per project and each of them should introduce new concepts / language features that weren't used in the last.
> 
> I think here you would want to stick with one project type, such as building up a web app. Reason being, if you've discussed a few concepts that are a bit dense without explanation, jumping to the data science section might bounce off it.

This is a valid point. Maybe the end goal should be a single thing. I think the focus should be web development as that's a fairly common niche that a lot of developers are in. Starting slow with scripts, then working on very basic console apps, but the end goal should be a full-stack web application with F#. I'm thinking Giraffe and Feliz/Sutil with Elmish.

**Comment by RonaldTM9999** [comment]
> > We should start slow with console applications and eventually progress to web applications, data science, and other concepts. 1 chapter per project and each of them should introduce new concepts / language features that weren't used in the last.
> 
> I think here you would want to stick with one project type, such as building up a web app. Reason being, if you've discussed a few concepts that are a bit dense without explanation, jumping to the data science section might bounce off it.

I agree. I would take a step further to suggest that the focus should be on teaching the domain and using F#, libraries and frameworks as teaching technology. If it's web development for example, I'd suggest introducing people to HTML, CSS, web storage, web security, etc. Assume the lowest common denominator because if they find the sit and are new to the domain as well, they will find a resource in language X's ecosystem that introduces them to the domain. They will argue that language X is easier to learn and tell everyone that and then stick to it because they then see there are job opportunities available for it. That becomes a double-count against F# and leaves us looking experimental academic hobbyists.
